Lab_3

vitttoria opened the pull request.

vitttoria committed:
Lab_3

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,110 @@
+            raise TypeError("Длительность книги - число с плавающей запятой")
+
+
+def __str__():
```
Непонятные функции, вынесены за класс

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,110 @@
+        # self.author = author
+        self.get_pages(pages)  # атрибут подкласса "Бумажная книга"
+
+    def get_pages(self, pages: int):
```
Ничего не возвращается, хотя по названию кажется что должно вернутся число страниц

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,110 @@
+        # self.author = author
+        self.get_duration(duration)
+
+    def get_duration(self, duration: float):
```
проверки должны быть через свойства
`@property.setter`

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,110 @@
+             Примеры:
+             # >>> book = Book("I am legend", 'John C.')  # инициализация экземпляра класса
+             """
+        self._name = name
```
как было описано в задании name и author должны быть объявлены через свойство (`@property`)

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,110 @@
+        super().__init__(name, author)
+        # self.name = name
+        # self.author = author
+        self.get_pages(pages)  # атрибут подкласса "Бумажная книга"
```
pages должно иметь свойство для установки (`@property.setter`) посмотрите на практике и лекции это приводилось

EgorOb reviewed (changes requested):
Много замечаний, нужно поправить хотя бы со свойствами (`@property`) разобраться

vitttoria committed:
Update task1.py

Исправила часть замечаний. Не понимаю, почему не получается для pages cделать @property.setter. Выдает ошибку: 
"line 44, in pages
    return self.pages
  [Previous line repeated 993 more times]
RecursionError: maximum recursion depth exceeded"

 Для duration сделала по-другому.

EgorOb reviewed (approved):
Зачтено

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,115 @@
+             Примеры:
+             # >>> book1 = PaperBook("I am legend", 'John C.', 300)  # инициализация экземпляра класса
+             """
+        super().__init__(name, author)
```
pages не определено при инициализации
нужно сделать 
`self.pages = pages`

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,115 @@
+
+    @property
+    def pages(self):
+        return self.pages
```
тут тогда можно сделать защищенный атрибут
`return self._pages`

EgorOb commented on Lab_3/task1.py:
```diff
@@ -0,0 +1,115 @@
+                :raise ValueError: Если значение меньше 0, то вызывается ошибка
+                Примеры:
+                # >>> paper_book1 = PaperBook('Yellow', 'Martin Cook', 6)
+                """
```
а тут его установить
```python
        if not isinstance(pages, int):
            raise TypeError("Количество страниц должно быть целым")
        if pages < 0:
            raise ValueError("Количество страниц должно быть положительным")
        self._pages = pages
```
